src/components/SQFormDialog/SQFormDialogInner.js

```diff
@@ -45,49 +45,76 @@ function SQFormDialogInner({
   const actionsClasses = useActionsStyles();
   const {resetForm} = useSQFormContext();
 
-  const handleClose = () => {
+  const [
+    isDialogAlertOpen,
+    {openDialog: openDialogAlert, closeDialog: closeDialogAlert}
+  ] = useDialog();
+
+  const handleCancel = () => {
+    openDialogAlert();
```

Comment:
Are you popping the modal open every time Cancel is clicked or only if the form is dirty?  I think we want to only do open the dialog if the form is dirty.

Reply:
This is still a blocker from my first review @shahzaib-haider1 
Your other changes look really good. From what I see, this is the last remaining issue.

The AYS modal should only open IF changes to the form have been made
Otherwise, the cancel button closes the modal as usual.

Reply:
@SeanGroff I have updated the PR and now AYS modal will be open when the form is dirty